### PR TITLE
Added the option to delete empty directories [#1206]

### LINKED ISSUE #1206: Empty directories can be removed when the library is consolidated.
As an administrator, I would like to have empty directories removed when I consolidate my library. This is a feature I want to turn on and off from the library configuration  view.

### PR DESCRIPTION
**Pull request by mcpierce**
Closes #1206 

## Status
READY

## Does this PR contain migrations?
YES

## Before You Submit Your PR:
- [ ] Have you announced your PR on the comixed-dev mailing list?
- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactors existing code (code changes for efficiency or maintainability)
- [ ] Security fix (be sure to include the CVE in the commit message) 
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class DeleteEmptyDirectoriesProcessor implements ItemProcessor<File, Void> {
+  @Autowired private FileAdaptor fileAdaptor;
+
+  @Override
+  public Void process(final File directory) throws Exception {
+    if (directory.listFiles().length == 0) {
```
Comment (BRUCELLA2) [inline comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java]: Wouldn't this " ArrayUtils.isEmpty(directory.listFiles()) " be safer?

File: comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/readers/DeleteEmptyDirectoriesReader.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+@Component
+@Log4j2
+public class DeleteEmptyDirectoriesReader implements ItemReader<File> {
+  @Autowired private ConfigurationService configurationService;
+
+  List<File> directories;
+
+  @Override
+  public File read()
+      throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
```
Comment (BRUCELLA2) [inline comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/readers/DeleteEmptyDirectoriesReader.java]: If we throws Exception, we don't need to throw UnexpectedInputException, ParseException, NonTransientResourceException ?

**Review by BRUCELLA2 — CHANGES_REQUESTED** [review]

**Review by BRUCELLA2 — DISMISSED** [review]

**Review by BRUCELLA2 — COMMENTED** [review]
2 comments

File: comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java (hunk 1 above)
Comment (mcpierce) [reply comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java]: I'm not sure, TBH. Is there a risk of File.listFiles() returning a null value? I had thought an instance of File will return an empty array if there are no contents. 

But, that said, I like your suggestion better. It makes it clear and readable the goal of the call. I'll update tonight and push the change you suggest.

File: comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/readers/DeleteEmptyDirectoriesReader.java (hunk 2 above)
Comment (mcpierce) [reply comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/readers/DeleteEmptyDirectoriesReader.java]: True. I'll clean that up as well.

File: comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java (hunk 1 above)
Comment (BRUCELLA2) [reply comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java]: In the doc, they say:
Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.
But I was thinking of making the change for better readability of the purpose of the code.

**Review by BRUCELLA2 — APPROVED** [review]

**Pull request merged**